app/controllers/bex_controller.rb

```diff
@@ -2,23 +2,33 @@ class BexController < ApplicationController
   before_action :authenticate_user!
 
   def agenda_jap
-    current_date = params.key?(:date) ? params[:date] : Date.today.next_occurring(:friday)
-
-    @appointments = policy_scope(Appointment).for_a_date(current_date).active
+    @appointment_type = AppointmentType.find_by(name: "Sortie d'audience SAP")
+    @current_date = current_date(@appointment_type, params)
+    @agendas = Agenda.in_organization(current_organization).with_open_slots_for_date(@current_date, @appointment_type)
```

Comment:
On pourrait virer partout la logique `in_organization` in me semble. On peut créer une has_many through entre organization et agenda et faire des `current_organization.agendas` à la place